source/kraken/tests/realtime_test.cpp

```diff
@@ -2740,3 +2740,299 @@ BOOST_AUTO_TEST_CASE(should_get_correct_base_stop_time_with_lollipop_II) {
+    BOOST_CHECK_EQUAL(vj->adapted_validity_pattern()->days, year("11111111"));
+    BOOST_CHECK_EQUAL(vj->rt_validity_pattern()->days, year("11111111"));
+
+    vj = pt_data.vehicle_journeys_map["vj_new_trip:modified:0:feed-1"];
+    BOOST_CHECK_EQUAL(vj->uri, "vj_new_trip:modified:0:feed-1");
+    BOOST_CHECK_EQUAL(vj->idx, 1);
+    BOOST_CHECK_EQUAL(vj->meta_vj->get_label(), "vj_new_trip");
+    BOOST_CHECK_EQUAL(vj->stop_time_list.size(), 4);
+    BOOST_CHECK_EQUAL(vj->base_validity_pattern()->days, year("00000000"));
+    BOOST_CHECK_EQUAL(vj->adapted_validity_pattern()->days, year("00000000"));
+    BOOST_CHECK_EQUAL(vj->rt_validity_pattern()->days, year("00000001"));
+
+    // New trip added
+    res = compute("20190101T073000", "stop_point:A", "stop_point:G");
+    BOOST_CHECK_EQUAL(res.response_type(), pbnavitia::ITINERARY_FOUND);
+    BOOST_CHECK_EQUAL(res.journeys_size(), 1);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times_size(), 4);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(0).departure_date_time(), "20190101T080000"_pts);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(0).stop_point().uri(), "stop_point:A");
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(1).departure_date_time(), "20190101T083000"_pts);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(1).stop_point().uri(), "stop_point:E");
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(2).departure_date_time(), "20190101T090000"_pts);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(2).stop_point().uri(), "stop_point:F");
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(3).departure_date_time(), "20190101T093000"_pts);
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).stop_date_times(3).stop_point().uri(), "stop_point:G");
+    BOOST_CHECK_EQUAL(res.journeys(0).sections(0).type(), pbnavitia::PUBLIC_TRANSPORT);
+    // impact
+    BOOST_CHECK_EQUAL(res.impacts_size(), 1);
+    BOOST_CHECK_EQUAL(res.impacts(0).uri(), "feed-1");
+    BOOST_CHECK_EQUAL(res.impacts(0).severity().effect(), pbnavitia::Severity_Effect_ADDITIONAL_SERVICE);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects_size(), 1);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops_size(), 4);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(0).departure_status(), pbnavitia::StopTimeUpdateStatus::ADDED);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(1).departure_status(), pbnavitia::StopTimeUpdateStatus::ADDED);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(2).departure_status(), pbnavitia::StopTimeUpdateStatus::ADDED);
+    BOOST_CHECK_EQUAL(res.impacts(0).impacted_objects(0).impacted_stops(3).departure_status(), pbnavitia::StopTimeUpdateStatus::ADDED);
+
+
+
+
+    // Recall the same new trip.
```

Comment:
You have to see _test logs_ to glimpse the behavior
```
...
[19-02-26 15:15:38,317] [TRACE] [] - realtime trip update received realtime.cpp:537  
[19-02-26 15:15:38,317] [DEBUG] [] - Vehicle journey founded : vj_new_trip realtime.cpp:566  
[19-02-26 15:15:38,317] [DEBUG] [] - Creating disruption realtime.cpp:366  
[19-02-26 15:15:38,317] [DEBUG] [] - Deleting disruption: feed-1 apply_disruption.cpp:664  
[19-02-26 15:15:38,317] [DEBUG] [] - Deleting impact: feed-1 apply_disruption.cpp:652  
[19-02-26 15:15:38,317] [DEBUG] [] - we are going to cleanup the vj vj_new_trip:modified:0:feed-1 type.cpp:501  
[19-02-26 15:15:38,317] [DEBUG] [] - feed-1 deleted apply_disruption.cpp:655  
[19-02-26 15:15:38,317] [DEBUG] [] - disruption feed-1 deleted apply_disruption.cpp:675  
[19-02-26 15:15:38,317] [TRACE] [] - a trip has been changed realtime.cpp:421  
[19-02-26 15:15:38,317] [TRACE] [] - Adding stop time into impact realtime.cpp:424  
[19-02-26 15:15:38,317] [DEBUG] [] - Trip effect : additional service realtime.cpp:506  
[19-02-26 15:15:38,317] [DEBUG] [] - Disruption added realtime.cpp:525  
[19-02-26 15:15:38,317] [DEBUG] [] - applying disruption: feed-1 apply_disruption.cpp:681  
[19-02-26 15:15:38,317] [TRACE] [] - Adding impact: feed-1 apply_disruption.cpp:517  
[19-02-26 15:15:38,317] [TRACE] [] - Start to add impact feed-1 on object vj_new_trip apply_disruption.cpp:116  
[19-02-26 15:15:38,317] [TRACE] [] - modifying vj_new_trip apply_disruption.cpp:206 
...
```
the function that do this job
```
void delete_disruption(const std::string& disruption_id, ...) apply_disruption.cpp:660
```

Reply:
As @kadhikari said, we want to see into the test a difference, to show if the second GTFS-RT replace the first.
To do that, I change in the second GTFS-RT only the stop time schedule.